vortex-fastlanes/src/bitpacking/compute.rs

```diff
@@ -52,31 +54,48 @@ impl ScalarAtFn for BitPackedArray {
 impl TakeFn for BitPackedArray {
     fn take(&self, indices: &dyn Array) -> VortexResult<ArrayRef> {
         let prim_indices = flatten_primitive(indices)?;
-        // Group indices into 1024 chunks and relativise them to the beginning of each chunk
+
+        // Group indices into 1024-element chunks and relativise them to the beginning of each chunk
         let relative_indices: Vec<(usize, Vec<u16>)> = match_each_integer_ptype!(prim_indices.ptype(), |$P| {
-            let grouped_indices = prim_indices
+            prim_indices
                 .typed_data::<$P>()
                 .iter()
-                .group_by(|idx| (**idx / 1024) as usize);
-            grouped_indices
+                .sorted()
```

Comment:
This won't work - you have to return results in the same order they have been asked for

Reply:
so as you deduplicate you need to create another list of indicies in the deduplicated list. i.e. if you have `[2, 2000, 2, 2000]` I'd produce something like `[(0, [2]), (1, [976])]` and `[(0, 0), (1, 0), (0, 0), (1, 0)]`.

Reply:
I wonder if that's worth the complexity, or if we just accept that out of order takes will just go down the "singles" path?